src/agents/common/agent.py

```diff
@@ -141,16 +141,17 @@ def _subtask_selector_node(self, state: BaseAgentState) -> dict[str, Any]:
         }
 
     async def _invoke_chain(self, state: BaseAgentState, config: RunnableConfig) -> Any:
-        inputs = {
-            AGENT_MESSAGES: state.get_agent_messages_including_summary(),
-            "query": state.my_task.description,
-        }
+        input_messages = state.get_agent_messages_including_summary()
         if len(state.agent_messages) == 0:
-            inputs[AGENT_MESSAGES] = filter_messages(state.messages)
+            input_messages = filter_messages(state.messages)
 
+        # invoke the chain.
         response = await ainvoke_chain(
             self.chain,
-            inputs,
+            {
+                AGENT_MESSAGES: filter_valid_messages(input_messages),
+                "query": state.my_task.description,
```

Comment:
What about the case that `description` is `None`?
```
    my_task: SubTask | None = None
```

Reply:
The agent will only be invoked when the subtask is not None.